# Genome-scale metabolic models for natural and long-term drug-induced viral control in HIV-infection
Genome-scale metabolic models (GSMMs) can provide novel insights into metabolic reprogramming during disease progression and therapeutic interventions. We developed a context-specific system-level GSMM of people living with HIV (PLWH) using global RNA sequencing data from peripheral blood mononuclear cells (PBMCs) with suppressive viremia either by natural (elite controllers, PLWHEC) or drug-induced (PLWHART) control. This GSMM was compared with HIV-negative controls (HC) to provide a comprehensive systems-level metabo-transcriptomic characterization. Transcriptomic analysis identified upregulation of oxidative phosphorylation (OXPHOS) as a characteristic of PLWHART, differentiating them from PLWHEC with dysregulated complexes I, III, and IV. The flux balance analysis identified altered flux in several intermediates of glycolysis including pyruvate, Î±-ketoglutarate, and glutamate, among others, in PLWHART. The in vitro pharmacological inhibition of OXPHOS complexes in a latent lymphocytic cell model (J-Lat 10.6) suggested a role for complex IV in latency reversal and immunosenescence. Furthermore, inhibition of complexes I/III/IV induced apoptosis, collectively indicating their contribution to reservoir dynamics.
